test/e2e/k8s_util.go

```diff
@@ -93,7 +93,13 @@ type probeResult struct {
 func (k *KubernetesUtils) GetPodByLabel(ns string, name string) (*v1.Pod, error) {
 	pods, err := k.getPodsUncached(ns, "pod", name)
 	if err != nil || len(pods) == 0 {
-		return nil, errors.WithMessagef(err, "unable to get Pod in Namespace %s with label pod=%s", ns, name)
+		errMsg := fmt.Sprintf("unable to get Pod in Namespace %s with label pod=%s", ns, name)
+		if err != nil {
+			return nil, errors.Wrap(err, errMsg)
+		}
+		if len(pods) == 0 {
+			return nil, errors.New(errMsg)
+		}
```

Comment:
`KubernetesUtils.waitForPodInNamespace` is affected by the change and needs update. It kind of relied on the bug.
And to make code cleaner, there may be no reason to still keep L95 given you have checked the two conditions separately anyway. Consider this:
```
var ErrPodNotFound = errors.New("Pod not found")

...
pods, err := k.getPodsUncached(ns, "pod", name)
if err != nil {
        return nil, errors.WithMessagef(err, "unable to get Pod in Namespace %s with label pod=%s", ns, name)
}
if len(pods) == 0 {
        return nil, ErrPodNotFound
}
return &pods[0], nil

...
func (k *KubernetesUtils) waitForPodInNamespace(ns string, pod string) ([]string, error) {
	log.Infof("Waiting for Pod '%s/%s'", ns, pod)
	for {
		k8sPod, err := k.GetPodByLabel(ns, pod)
		if err != nil && err != errPodNotFound {
			return nil, errors.WithMessagef(err, "unable to get Pod '%s/%s'", ns, pod)
		}

		if k8sPod != nil && k8sPod.Status.Phase == v1.PodRunning {
                ...
```